File: Editorial/0151_Reverse_Words_in_a_String/Go/1_Built_in_Split_Reverse.go
func reverseWords(s string) string {
    // split by spaces
    words := strings.Fields(s)
    // reverse
    for i := 0; i < len(words)/2; i++ {
        words[i], words[len(words)-i-1] = words[len(words)-i-1], words[i]
    }
    // join the words with a space
    return strings.Join(words, " ")
}

File: Editorial/0151_Reverse_Words_in_a_String/CSharp/1_Built_in_Split_Reverse.cs
public class Solution {
    public string ReverseWords(string s) {
        // remove leading and trailing spaces
        s = s.Trim();
        // split by spaces and reverse
        string[] words = s.Split(new char[] { ' ' },
            StringSplitOptions.RemoveEmptyEntries);
        Array.Reverse(words);
        // join the words with a space
        return String.Join(" ", words);
    }
}
